### ISSUE TITLE
[BUG] 'RuntimeError: dictionary changed size during iteration' in '_generalized_statsforecast.py'

### ISSUE THREAD

**Issue by corradomio** [sktime/sktime#5449]
**Describe the bug**
During the training of 'sktime.forecasting.statsforecast.StatsForecastAutoARIMA' model it is raised the exception

'RuntimeError: dictionary changed size during iteration'

The error, in the code, is obvious:
line 53 of '_generalized_statsforecast.py':

the dictionary ' sktime_params' is scanned and if it is found a 'strange' parameter, it is removed with 'sktime_params.pop(...)'
BUT this changes the dictionary!

Solution: to use 'list(sktime_params.keys())': this collects the keys in a list and the dictionary can be changed without problems

```python
    def _get_validated_statsforecast_params(self):
        sktime_params = self._get_statsforecast_params()
        sktime_default_params = self.get_param_defaults().keys()
        statsforecast_params = self._get_init_statsforecast_params()

        # [CM] the dictionary 'sktime_params' is changed with 'sktime_params.pop(...)'
        # add 'list()'
        # for sktime_param in sktime_params.keys():
        for sktime_param in list(sktime_params.keys()):
            if sktime_param not in statsforecast_params:
                sktime_params.pop(sktime_param)
                if sktime_param not in sktime_default_params:
                    warn(
                        f"Keyword argument '{sktime_param}' will be omitted as it is"
                        f" not found in the __init__ method "
                        f"from {self._get_statsforecast_class()}. "
                        f"Check your statsforecast version"
                        f"to find out the right API parameters."
                    )
        return sktime_params
```

**Versions**
Python dependencies:
          pip: 23.2.1
       sktime: 0.23.0
      sklearn: 1.3.1
       skbase: 0.5.1
        numpy: 1.25.2
        scipy: 1.11.2
       pandas: 2.1.1
   matplotlib: 3.8.0
       joblib: 1.3.2
        numba: 0.58.0
  statsmodels: 0.14.0
     pmdarima: 2.0.3
statsforecast: 1.6.0
      tsfresh: None
      tslearn: None
        torch: 2.0.1+cpu
   tensorflow: None
tensorflow_probability: None

**Comment by corradomio** [comment]
It seems resolved in 0.24.0

**Comment by fkiraly** [comment]
Yes, this is the fix: https://github.com/sktime/sktime/pull/5317
